### PR TITLE
Adds the 'Genderless' option to all species

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

Adds the 'Genderless' gender option to all species, rather than just some of them. This changes all pronouns when referring to the character to "They" rather than "He" or "She".
Also adds a 'Body' preference to the character creator, which controls the character's gender sprite (if applicable).
With the new options, body type (male or female sprite) is controlled with the 'Body' setting, and ingame gender (pronoun, emotes, etc.) is set with the usual 'Gender' setting.

*(In response to https://www.paradisestation.org/forum/topic/20709-character-options)*

## Why It's Good For The Game

Diona, Drask, Greys, IPCs, and Vox all already have this as a character option. And I'd imagine that in the year 2565 *any* species being genderless/nonbinary would be pretty normal, considering that it already is in 2021.
I could go into detail about why this would be good to have, but I honestly don't know much about the whole 'gender politics' thing. I just think that people feeling more comfortable roleplaying as their character (or just playing in general really) is a good thing.

## Images of changes

new

(Note "They" rather than "He" or "She".)

## Changelog
:cl:
add: Added the 'Genderless' gender option for all species, as well as a new 'Body' option for choosing the sprite variant separately.
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Putting my post from the forums here
```
Male and female human sprites are different, so this would require a new set of sprites.

 

Now for the bit where twitter cancels me.

The gender option determines how your body is ingame. IRL, you have either a male body or a female body. Thats how it is.

This does raise the question of "Add a separate preference as what you want to identify as", but that would require redoing of all the existing pronoun code to use an identify_gender var instead of the regular one, which is just going to cause headaches all around.
```

TLDR: Not in favour.

**Comment by Samoa7** [comment]
> is it that hard to just not dip into irl stuff for a code issue

**Comment by SabreML** [comment]
It *does* use the female body sprites for genderless at the moment, but I didn't think that would be too much of an issue since all the species who have the system already have the exact same thing. (I'd be happy to make some new sprites for it if nessecary.)
This is far more for the pronouns than anything else, and to allow people to play as a character more in line with their own personality.

Obviously at the moment it works similar to IRL, you can't know just by looking at someone what they'd like to be referred to as, but this is an online sci-fi roleplaying game set 500 years in the future. Some liberties can be taken here in that respect, or at least I think so.

**Comment by The-Sun-In-Splendour** [comment]
This is far more for the pronouns than anything else, and to allow people to play as a character more in line with their own personality.


Allow people to play more accurate self inserts? That's what I'm understanding from what you're saying.

**Comment by FlattestGuitar** [comment]
> but that would require redoing of all the existing pronoun code to use an identify_gender var instead of the regular one

You don't have to sacrifice anything to add this to the code.

You already have the `PLURAL` gender used very widely. Pronouns already support this. I'm not sure why you said they don't.

As for the sprites, just add a separate sprite preference for when the species has multiple sprite sheets and go wild. You could probably even just make a different preference create a different organ to remove genders from sprite drawing altogether, which is just cleaner code.

Going `Now before twitter cancels me` sounds mean spirited. Many members of this community play it as a form of escapism and Paradise itself was built to be a safe and accepting space. You have an enormous share of LGBT+ folks here who can just be who they want to be, which for many of us is not possible in real life.

Please don't be dismissive of this, @AffectedArc07

**Comment by Cocacolagua** [comment]
> It _does_ use the female body sprites for genderless at the moment, but I didn't think that would be too much of an issue since all the species who have the system already have the exact same thing. (I'd be happy to make some new sprites for it if nessecary.)

@SabreML I believe nobody cares on those scenarios due to those species being alien-looking/hard-to-distinguish-between-bodies, it would be better to allow choosing which body-type.

**Comment by SabreML** [comment]
> @SabreML I believe nobody cares on those scenarios due to those species being alien-looking/hard-to-distinguish-between-bodies, it would be better to allow choosing which body-type.

Good point.
Having separate options for body sprite and pronoun would solve a lot of issues, but I'd probably need to add a new DB setting for it and those are frozen ATM. And even as the person who made this PR I feel like having two separate options for gender might be a bit too much to add in one go.
That said, this isn't exactly going to get speedmerged anyway so I guess I can add it if people are in favour.

**Comment by DogsParadise** [comment]
How am I gonna phrase this politically correct without being cancelled..
Let me try.

Personally, I don't mind how anyone defines as or wishes to be defined as.
But we are playing a game here as a male, I can play a female if I so wish to.
Nobody would care.

I have two issues however with this PR:
It's going to promote a type of RP which I feel is just going to be silly. That already.. happens enough.
Me: He *refering to someone* can probably grab you the toolbox
Them: I AM NOT A HE, I am a it/they/helicopter/anything/etc/etc/etc/

Besides, this is only going to add bloat to sprites if it gets added as a seperate sprite.

In no way does this compare to vox and other species, those are aliens. 
Vox are called Vox, not he/she.

**Comment by FreeStylaLT** [comment]
:trollface:

**Comment by Varghh** [comment]
@FlattestGuitar already said everything I think.
I don't see a single reason why this should not be added into the game, our players are the server, giving them more options to play their characters the way they want sound like the way to go.

**Comment by Sirryan2002** [comment]
Implementing this PR would add the option for people to choose a gender via a feature that is already in place and dissing/supporting this PR for things related to how sprites are handled in relation to being genderless is out of scope for this PR and can be saved for future discussion down the road. 

I agree with @FlattestGuitar in the sense that this does not hurt our code in any way shape or form and given our community I believe this a QOL change for many people and would serve to add to their experience only positively. As long as there are not outstanding long term code issues CAUSED(not other PRs that will arise b/c of this) by this, the only determinant for merge/closing should be on whether or not this change is right for our community and I think that's pretty clear already, I hope people can place political opinions for this aside and take a more wholistic look at the benefits/cons of this PR for our players.

**Comment by SabreML** [comment]
I get where you're coming from with these, but personally I disagree. (Which I guess is a given since I'm the one who made the PR.)
> I have two issues however with this PR:
> It's going to promote a type of RP which I feel is just going to be silly. That already.. happens enough.
> Me: He _refering to someone_ can probably grab you the toolbox
> Them: I AM NOT A HE, I am a it/they/helicopter/anything/etc/etc/etc/

I very much doubt this will happen. Five species already have a genderless option and I've never once seen any of them complain about being called the wrong thing. On top of that, I see people calling female characters "he" at *least* twice a round, but aside from the occasional correction I've not seen any issues.

> Besides, this is only going to add bloat to sprites if it gets added as a seperate sprite.

That's true, although it's only one of the possible options if the body sprite turns out to be a sticking point.
Even if that *is* the way it ends up going, I personally consider three extra sprites in each file worth the tradeoff.

> P.S. I am ready for the cancelling and being called names for disagreeing with a pr

People are allowed to have their own opinions on something, this isn't an echo chamber despite how it can seem sometimes.
If someone has an issue with a PR then I definitely encourage them to say so, especially with this one considering the subject matter.

The only thing I will say though is to please try to keep any politics out of it. This is a pull request adding a pronoun option, *not* a discussion forum about the merits of gender identity.

**Comment by AffectedArc07** [comment]
Decision made between the maintainers to lock this before it turns into a political wasteland.

File: SQL/paradise_schema.sql [hunk 1]
```diff
@@ -27,6 +27,7 @@ CREATE TABLE `characters` (
   `real_name` varchar(45) COLLATE utf8mb4_unicode_ci NOT NULL,
   `name_is_always_random` tinyint(1) NOT NULL,
   `gender` varchar(11) COLLATE utf8mb4_unicode_ci NOT NULL,
+  `body_type` varchar(11) COLLATE utf8mb4_unicode_ci NOT NULL,
```
Comment (AffectedArc07) [inline SQL/paradise_schema.sql:30]: Need to update prefixed schema

File: code/modules/client/preference/preferences_mysql.dm [hunk 2]
```diff
@@ -206,50 +207,51 @@
...
+		h_style = query.item[17]
+		f_style = query.item[18]
+		m_styles = params2list(query.item[19])
+		ha_style = query.item[20]
+		alt_head = query.item[21]
+		e_colour = query.item[22]
+		underwear = query.item[23]
+		undershirt = query.item[24]
+		backbag = query.item[25]
+		b_type = query.item[26]
```
Comment (AffectedArc07) [inline code/modules/client/preference/preferences_mysql.dm:232]: Absolutely not.

Put the column for this at the end of the select statement to avoid creating conflict and problem hell here

File: code/modules/client/preference/preferences.dm [hunk 3]
```diff
@@ -1976,23 +1994,6 @@ GLOBAL_LIST_INIT(special_role_times, list( //minimum age (in days) for accounts
...
-							if("Female")
-								gender = FEMALE
-							if("Genderless")
-								gender = PLURAL
-					else
-						if(gender == MALE)
-							gender = FEMALE
-						else
-							gender = MALE
-					underwear = random_underwear(gender)
```
Comment (AffectedArc07) [inline code/modules/client/preference/preferences.dm:1994]: Someone else can look this part over, I dont trust my own review on mob code

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
You also need to update all the configs to ensure default DB is version 23.

While I am here, some other points:
- We have multiple DB PRs open already, with this one being the lowest priority
- After discussing with some other people, I still ain't onboard with this, because the minority will ruin it for the majority with "DID YOU JUST CALL ME A HE OH MY GOD", which is just going to make people feel worse over things due to a blatant pisstake.

Oh, and if people start berating me over my decision here, that's just going to make me even less likely to consider this because I am not about to promote a political wasteland. 

Again, if you turn this PR (Or discussion surrounding it on the forums or discord) into a political wasteland, we will do something about it.

**Comment by AffectedArc07** [comment]
This PR has been unlocked to allow the author to edit the description as the PR has changed. **Do not add comments to this PR**

Anyone else adding to or altering this discussion will be met with gitbans.

**Comment by AffectedArc07** [comment]
This PR has been open for over a week and a half with no one in favour from either party, and 3 objections from both parties combined. 

We see this implementation as causing a lot of headaches down the road due to the mix of body_type and the built in BYOND gender var that is on all atoms. We also believe that this is just going to encourage less-than-favourable behaviour and could end up causing legitimate offence to those who personally identify this way.

We are closing this discussion. Anyone trying to bring this discussion out of context and attempt to gaslight leadership will not be tolerated.
